components/scream/src/physics/p3/p3_functions_ice_melting_impl.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef P3_FUNCTIONS_ICE_MELTING_IMPL_HPP
+#define P3_FUNCTIONS_ICE_MELTING_IMPL_HPP
+
+#include "p3_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace p3 {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::ice_melting(const Spack& rho, const Spack& t, const Spack& pres, const Spack& rhofaci,
+	      const Spack& f1pr05, const Spack& f1pr14, const Spack& xxlv, const Spack& xlf, 
+	      const Spack& dv, const Spack& sc, const Spack& mu, const Spack& kap, 
+	      const Spack& qv, const Spack& qitot_incld, const Spack& nitot_incld,
+	      Spack& qimlt, Spack& nimlt)
+{
+  // Notes Left over from WRF Version:
+  // need to add back accelerated melting due to collection of ice mass by rain (pracsw1)
+  // note 'f1pr' values are normalized, so we need to multiply by N
+  // currently enhanced melting from collision is neglected
+  // include RH dependence
+
+  //Find cells above freezing AND which have ice
+  const auto has_melt_qi = (qitot_incld >= C::QSMALL ) && (t > C::Tmelt);
+
+  if (has_melt_qi.any()){
+
+    //PMC qv_sat from math_impl.hpp seems to match hardcoded formula from F90 I'm swapping in C++ ver.
+    //    Note that qsat0 should be with respect to liquid. Confirmed F90 code did this.
+    //PMC C::Tmelt is scalar and pres is Spack... does Spack(C::Tmelt) work correctly?
+    const auto qsat0 = qv_sat(Spack(C::Tmelt), pres, false); //last false means NOT saturation w/ respect to ice.
+      
```

Comment:
In the line above, Tmelt is a scalar and pres is a Spack. qv_sat expects both inputs to be the same size. I'm trying to copy the scalar value of Tmelt to all entries in a Spack-sized temperature array here and guessed that Spack(scalar) would do the trick. I'm not sure whether this is the case and I struggled to look through share/ code's Spack functions to figure this out. Hints?

Reply:
Yes: https://github.com/E3SM-Project/scream/blob/df801e5cc3ffa0d1b77e6fb90708fddd0ebcdd0c/components/scream/src/share/scream_pack.hpp#L160

Reply:
Happy days. Thanks.